src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/RadioactivityTask.java

```diff
@@ -0,0 +1,160 @@
+import io.github.thebusybiscuit.slimefun4.core.attributes.ProtectionType;
+import io.github.thebusybiscuit.slimefun4.core.attributes.Radioactive;
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
+import io.github.thebusybiscuit.slimefun4.implementation.items.RadioactiveItem;
+import io.github.thebusybiscuit.slimefun4.utils.itemstack.ItemStackWrapper;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
+import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
+import net.md_5.bungee.api.ChatMessageType;
+import net.md_5.bungee.api.chat.ComponentBuilder;
+import org.bukkit.Bukkit;
+import org.bukkit.GameMode;
+import org.bukkit.Material;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.potion.PotionEffect;
+import org.bukkit.potion.PotionEffectType;
+
+import javax.annotation.Nonnull;
+import java.util.HashMap;
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ * The {@link RadioactivityTask} handles radioactivity for
+ * {@link Radioactive} items.
+ *
+ * @author Semisol
+ *
+ */
+public class RadioactivityTask implements Runnable {
+    private static final Symptom[] SYMPTOMS = Symptom.values();
+    private static final HashMap<UUID, Integer> radioactivityLevel = new HashMap<>();
+    private final int duration = SlimefunPlugin.getCfg().getOrSetDefault("options.radiation-update-interval", 1) * 20 + 20;
+    private final PotionEffect WITHER = new PotionEffect(PotionEffectType.WITHER, duration, 1);
+    private final PotionEffect WITHER2 = new PotionEffect(PotionEffectType.WITHER, duration, 4);
+    private final PotionEffect BLINDNESS = new PotionEffect(PotionEffectType.BLINDNESS, duration, 0);
+    private final PotionEffect SLOW = new PotionEffect(PotionEffectType.SLOW, duration, 3);
+
+    public static void removePlayer(@Nonnull Player p){
+        radioactivityLevel.remove(p.getUniqueId());
+    }
+
+    @Override
+    public void run() {
+        for (Player p : Bukkit.getOnlinePlayers()) {
+            if (!p.isValid() || p.isDead()) {
+                continue;
+            }
+
+            PlayerProfile.get(p, profile -> handleRadiation(p, profile));
+        }
+    }
+
+    private void handleRadiation(@Nonnull Player p, @Nonnull PlayerProfile profile) {
+        if (p.getGameMode() == GameMode.CREATIVE || p.getGameMode() == GameMode.SPECTATOR) {
+         return;
+        }
+        Set<SlimefunItem> radioactiveItems = SlimefunPlugin.getRegistry().getRadioactiveItems();
+
+        int exposureTotal = 0;
+        UUID uuid = p.getUniqueId();
+
+        if (!profile.hasFullProtectionAgainst(ProtectionType.RADIATION)) {
+            for (ItemStack item : p.getInventory()) {
+                if (item == null || item.getType() == Material.AIR) {
+                    continue;
+                }
+                ItemStack tmpItem = item;
+
+                if (!(item instanceof SlimefunItemStack) && radioactiveItems.size() > 1) {
```

Comment:
This is not used anymore because you cast the RadioactiveItem instead of searching for it like before.
Also, you dont need to load the radioactiveItems as far as I can see.

Reply:
Wouldnt switch around the if statements speed up the if a little bit aswell?

Reply:
This whole if statement can be removed, the reason why it was added is already irrelevant.